### ISSUE TITLE
fatal problem with saving variables

### ISSUE THREAD

**Issue by ad26kt** [tensorflow/tensorflow#10074]
I coded a simple feedforward neural network and it works very well.

I tried to save the computation time, and created:
**self.total_time** = tf.Variable(0, dtype = tf.float32, trainable = True, name = 'total_time')
in the fnn class.

and i tried to print the total training time per some training epoch.
**I made it to grow with time:**

**# Check & Print training time**
till_now = time.time() - start_time
self.total_time += till_now
print_time(self.total_time.eval())

**and the result look something like this :** 

Epoch :   0 | Evaluation :  115 | Learning Rate : 0.50
-------------------------------------------------------
Training Loss :         0.040919
Validation Loss :      0.0741969
Validation Accuracy :      97.77%
Total time cost : 0.38 seconds

Epoch :   1 | Evaluation :  116 | Learning Rate : 0.50
-------------------------------------------------------
Training Loss :        0.0417941
Validation Loss :       0.073841
Validation Accuracy :      97.73%
Total time cost : 0.71 seconds

Epoch :   2 | Evaluation :  117 | Learning Rate : 0.50
-------------------------------------------------------
Training Loss :        0.0334573
Validation Loss :      0.0745566
Validation Accuracy :      97.75%
Total time cost : 1.01 seconds

However, When i interrupt the training procedure and try to restore global variables and restart the training, I just lose the value of **variable total_time** and it initialized as 0 which is the value i first give to.
I also checked tf.global_variables() include self.total_time.

What is wrong?

**Comment by ad26kt** [comment]
I found that when **self.total_time += till_now**, the tensor name was changed.
it can be solved with assign function
